pkg/processor/controlcommunication/types.go

```diff
@@ -63,13 +63,34 @@ func (c *ControlConsumer) GetKind() ControlMessageKind {
 // Send broadcasts a message to all subscribed channels
 func (c *ControlConsumer) Send(message *ControlMessage) error {
 
+	wg := sync.WaitGroup{}
+	wg.Add(len(c.Channels))
 	for _, channel := range c.Channels {
```

Comment:
why is channel public? if we provide logic to delete/send and etc I think channel needs to be private

Reply:
make sense! changed